res/ari/resource_channels.c

```diff
@@ -1986,6 +2002,11 @@ void ast_ari_channels_dial(struct ast_variable *headers,
 		return;
 	}
 
+	if (!ast_strlen_zero(args->pre_dial_go_sub)) {
```

Comment:
I don't think this is safe either unless the channel lock is held for the entire duration, as the stasis thread has ownership over the channel as well.